### PR TITLE
Support For Location Entity Mapping In Sage Intacct

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: src/app/core/models/destination-attribute-detail.model.ts [hunk 1]
```diff
@@ -7,4 +7,5 @@ export type DestinationAttributeDetail = {
   gl_account_no: string;
   gl_account_title: string;
   is_reimbursable: boolean;
+  country?: string;
```
Comment (ashwin1111) [inline src/app/core/models/destination-attribute-detail.model.ts:10]: `?` id not needed

File: src/app/core/services/mappings.service.ts [hunk 2]
```diff
@@ -165,6 +165,18 @@ export class MappingsService {
     }));
   }
 
+  postLocationEntityMapping(locationEntityMappingPayload: any): Observable<any> {
```
Comment (ashwin1111) [inline src/app/core/services/mappings.service.ts:168]: let's not use `any` anywhere in the app. Please update with relevant model

File: src/app/si/settings/sage-intacct-configurations/location-entity/location-entity.component.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    that.isLoading = true;
+
+    const locationEntityMappingPayload: any = {
+      location_entity_name: siEntityMapping.value,
+      destination_id: siEntityMapping.destination_id,
+      country_name: siEntityMapping.detail.country,
+      workspace: that.workspaceId
+    };
+
+    that.mappingsService.postLocationEntityMapping(locationEntityMappingPayload).subscribe(() => {
```
Comment (ashwin1111) [inline src/app/si/settings/sage-intacct-configurations/location-entity/location-entity.component.ts:41]: once this is saved, let's call sync dimension. Or may be handle them in api via signals.

**Comment by ashwin1111** [comment]
@NileshPant1999 I don't see location entity stuff being removed from general mappings

File: src/app/si/settings/sage-intacct-configurations/sage-intacct-configurations.component.html [hunk 4]
```diff
@@ -6,6 +6,9 @@
       Configurations
     </div>
     <nav mat-tab-nav-bar>
+      <a mat-tab-link (click)="changeState('LOCATION_ENTITY')" [active]="state==='LOCATION_ENTITY'" *ngIf='sageIntacctConnectionDone'>
```
[the remark this answers is not in the document]
Comment (NileshPant1999) [reply src/app/si/settings/sage-intacct-configurations/sage-intacct-configurations.component.html:9]: no it is like that only see the below general tab

File: src/app/si/settings/sage-intacct-configurations/location-entity/location-entity.component.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      });
+    });
+  }
+
+  ngOnInit() {
+    const that = this;
+    that.workspaceId = that.route.snapshot.parent.parent.params.workspace_id;
+    that.isLoading = true;
+    that.mappingsService.getSageIntacctDestinationAttributes('LOCATION_ENTITY').subscribe((locationEntities) => {
+      that.siLocationEntities = locationEntities.filter(entity => entity.detail.country !== null);
```
[the remark this answers is not in the document]
Comment (NileshPant1999) [reply src/app/si/settings/sage-intacct-configurations/location-entity/location-entity.component.ts:69]: we need to add a default country for top-level

File: src/app/si/settings/sage-intacct-configurations/location-entity/location-entity.component.ts (hunk 3 above)
Comment (NileshPant1999) [reply src/app/si/settings/sage-intacct-configurations/location-entity/location-entity.component.ts:42]: @ashwin1111 do we really need to do this, as we are redirecting anyway to the dashboard page after saving location_entity and the sync is happening there, doing it via signal in the backend is i think not needed here, am i missing something here.

File: src/app/si/settings/sage-intacct-configurations/location-entity/location-entity.component.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  constructor(private formBuilder: FormBuilder,
+              private mappingsService: MappingsService,
+              private route: ActivatedRoute,
+              private router: Router) {}
+
+  submit() {
+    const that = this;
+
+    const locationEntityId = that.locationEntityForm.value.siLocationEntities;
+    const siEntityMapping = that.siLocationEntities.filter(filteredLocationEntity => filteredLocationEntity.destination_id === locationEntityId)[0];
```
[the remark this answers is not in the document]
Comment (NileshPant1999) [reply src/app/si/settings/sage-intacct-configurations/location-entity/location-entity.component.ts:32]: we need to have dummy destination id and dummy country for top-level location entity we store it in the destination attribute table via post signal in sage Intacct credentials.

File: src/app/si/settings/sage-intacct-configurations/sage-intacct-configurations.component.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/si/settings/sage-intacct-configurations/sage-intacct-configurations.component.html:9]: hey no, it's not the right convention. Also I am not sure if it takes it as a string. We shouldn't use this.

File: src/app/si/settings/sage-intacct-configurations/location-entity/location-entity.component.ts (hunk 3 above)
Comment (ashwin1111) [reply src/app/si/settings/sage-intacct-configurations/location-entity/location-entity.component.ts:42]: No, after you save sage intacct credentials we sync data currently. Can check the code and behavior.

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**